openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -201,3 +213,34 @@ func (e *extension) fetchLoggingHost(ctx context.Context) string {
 	}
 	return route.Status.Ingress[0].Host
 }
+
+func (e *extension) cleanupDomainMapping(ctx context.Context, ns string) error {
+	client := e.kubeclient
+	for _, dep := range []string{"domain-mapping", "domainmapping-webhook"} {
+		if err := client.AppsV1().Deployments(ns).Delete(ctx, dep, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete deployment %s: %w", dep, err)
+		}
+	}
+	// Delete the rest of the domain mapping resources
+	for _, svc := range []string{"domainmapping-webhook", "domain-mapping-sm-service", "domainmapping-webhook-sm-service"} {
+		if err := client.CoreV1().Services(ns).Delete(ctx, svc, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete service %s: %w", svc, err)
+		}
+	}
+	for _, lease := range []string{"domainmapping-webhook.defaultingwebhook.00-of-01", "domainmapping-webhook.validationwebhook.00-of-01",
```

Comment:
I think in this case to be 100% correct, you need to fetch all leases, filter the ones that are domain-mapping related and then remove only those.

Reply:
https://github.com/openshift-knative/serverless-operator/pull/2743/commits/cf5317c47b4d7d8551ed9839059f5a065c5003a6